### PR TITLE
Improvements in swupd_init

### LINKED ISSUE #551: swupd clean --all requires an internet connection
For when users are asking to remove all cached content, an internet connection should not be required.  Without one, the following message is seen:
```
Curl error: (5) Couldn't resolve proxy name
Curl error: (5) Couldn't resolve proxy name
Failed swupd initialization, exiting now
```
Not sure if the same would apply for `swupd clean` since it might try to optimize what it cleans up from checking some files online.

### PR DESCRIPTION
**Pull request by otaviobp**
- some clean up in swupd_init and swupd_deinit
 - Don't use curl on swupd-clean command (fixes #551)

### CODE REVIEW

File: src/lock.c [hunk 1]
```diff
@@ -89,7 +94,8 @@ int p_lockfile(void)
 }
 
 /* closes lock fd and must not unlink lock file (else race allowed) */
-void v_lockfile(int fd)
+void v_lockfile(void)
 {
-	close(fd);
+	if (lock_fd > 0)
```
Comment (matthewrsj) [inline src/lock.c]: braces around the if please (make compliant should be catching this)

File: src/lock.c [hunk 2]
```diff
@@ -38,13 +38,18 @@
...
+static int lock_fd = -1;
+
 /* Try to get a write lock region on the lock file. Returns:
 * >= 0 an fcntl region lock'd fd or exits with a positive error
 * code and a recommended course of action for user.
 */
 int p_lockfile(void)
 {
-	int lock_fd, ret;
+	if (lock_fd > 0)
```
Comment (matthewrsj) [inline src/lock.c]: braces

**Comment by otaviobp** [comment]
problems fixed

File: src/helpers.c [hunk 3]
```diff
@@ -521,27 +521,26 @@ void free_file_data(void *data)
...
 * initializations for the majority of commands, that is:
 * 	- Make sure root is the user running the code
-*	- Initialize log facility
-*	- Get the lock
+*       - Initialize globals
 *	- initialize mounted directories
+*       - Create necessary directories
+*	- Get the lock
 *	- Initialize curl
+*       - Initialize signature checking
```
Comment (castulo) [inline src/helpers.c]: Are those items in the comments with a different alignment on purpose? Are they like sub-bullets?

**Comment by otaviobp** [comment]
I sent a fix for the indent that @castulo commented (github hid the message)

**Pull request merged**